src/algebra/module/submodule/basic.lean

```diff
@@ -208,12 +208,19 @@ protected lemma nonempty : (p : set M).nonempty := ⟨0, p.zero_mem⟩
 
 @[simp] lemma mk_eq_zero {x} (h : x ∈ p) : (⟨x, h⟩ : p) = 0 ↔ x = 0 := subtype.ext_iff_val
 
+instance : coe_is_add_monoid_hom p M :=
+{ coe_zero := rfl,
+  coe_add := λ _ _, rfl }
+
+instance : coe_is_linear_map R p M :=
+{ coe_smulₛₗ' := λ _ _, rfl }
+
 variables {p}
 @[simp, norm_cast] lemma coe_eq_zero {x : p} : (x : M) = 0 ↔ x = 0 :=
 (set_like.coe_eq_coe : (x : M) = (0 : p) ↔ x = 0)
-@[simp, norm_cast] lemma coe_add (x y : p) : (↑(x + y) : M) = ↑x + ↑y := rfl
-@[simp, norm_cast] lemma coe_zero : ((0 : p) : M) = 0 := rfl
-@[norm_cast] lemma coe_smul (r : R) (x : p) : ((r • x : p) : M) = r • ↑x := rfl
+@[norm_cast] protected lemma coe_add (x y : p) : (↑(x + y) : M) = ↑x + ↑y := rfl
+@[norm_cast] protected lemma coe_zero : ((0 : p) : M) = 0 := rfl
+@[norm_cast] protected lemma coe_smul (r : R) (x : p) : ((r • x : p) : M) = r • ↑x := rfl
```

Comment:
I'd argue these are worth keeping as `simp` because they're true by `rfl` and therefore elegible for `dsimp`.